build.py

```diff
@@ -127,15 +131,36 @@ def _write_v2(self, outputdir):
 
     def _write_cluster_version_mapping(self, outputdir):
         srcpath = self.sourcedir / "templates_v2" / "cluster_version_mapping.json"
+        self._validate_cluster_version_mapping(srcpath)
         dstpath = outputdir / "cluster_version_mapping.json"
         logger.info(f"Writing cluster_version_mapping.json: {dstpath}")
         # preserve non-standard formatting of the file
         shutil.copy(srcpath, dstpath)
-        self._validate_cluster_version_mapping(dstpath)
 
     def _validate_cluster_version_mapping(self, filepath):
         content = self._load_json(filepath)
         self._assert_json_schema(filepath, content, "cluster_version_mapping.schema.json")
+        self._assert_cluster_version_mapping_order(filepath, content)
+
+    def _assert_cluster_version_mapping_order(self, filepath, content):
+        i = 0
+        v1 = semver.Version.parse(content[0][0])
```

Comment:
Maybe we can start with v1 being 0.0 and then make the loop through the content, rather than starting at index 1? I find it more readable, but it is very subjective, so whatever you prefer.

Reply:
I didn't want to initialize the variable with any specific version because it would put additional constraints on the file. `0.0.0-0` is a valid version for the first item in the file.

I somehow thought that starting with `v1 = None` would make the code more convoluted, but it actually worked out quite nice. I've pushed the change.

Reply:
OK, thank you.